lib/turn.rb

```diff
@@ -0,0 +1,92 @@
+class Turn
+  attr_reader :player1,
+              :player2,
+              :spoils_of_war
+    def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+  end
+
+  def type
+    if((@player1.deck.cards.length < 3) || (@player2.deck.cards.length < 3))
+      :sudden_death
+    else
+      if(@player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0))
+        :basic
+      # elsif((@player1.deck.cards.length < 3) || (@player2.deck.cards.length < 3))
+      #   :loss
+    elsif((@player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)) &&
+      (@player1.deck.rank_of_card_at(2) == @player2.deck.rank_of_card_at(2)))
+        :mutually_assured_destruction
+      elsif(@player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0))
+        :war
+      end
+    end
```

Comment:
Ideally, we would see `elsif` used instead of `else` with a nested `if`:
```
if (@player1.deck.cards.length < 3) || (@player2.deck.cards.length < 3)
  :sudden_death
elsif @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
  :basic
elsif ((@player1.deck.rank_of_card_at(0)) == (@player2.deck.rank_of_card_at(0))) && ((@player1.deck.rank_of_card_at(2)) == (@player2.deck.rank_of_card_at(2)))
  :mutually_assured_destruction
elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)
  :war
end
```